### ISSUE TITLE
🐛 Bug Report: Incorrect Path Style option for techdocs S3 publisher

### ISSUE THREAD

**Issue by chris-nolan** [backstage/backstage#15605]
### 📜 Description

Since upgrading from 1.8.3 to 1.9.1 publishing to an S3 compatible storage platform (minio) no longer works when using the `s3ForcePathStyle ` option in config (using `local` builder, `local` generator).

### 👍 Expected behavior

Should use path style access to S3 if option enabled.

I believe this is because, in the V3 library the option has changed name from `s3ForcePathStyle` to `forcePathStyle` as documented here https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-s3/interfaces/s3clientconfig.html#forcepathstyle whereas techdocs is using the old property name https://github.com/clareliguori/backstage/blob/cbef804d2375ec9fee75b2cde08397c7bbbcb143/plugins/techdocs-node/src/stages/publish/awsS3.ts#L156

### 👎 Actual Behavior with Screenshots

An error is generated on startup as the library is using the host/endpoint style of {bucketname}.{endpoint}, rather than {endpoint}/{bucketname}

### 👟 Reproduction steps

Startup backstage with the given style of config

### 📃 Provide the context for the Bug.

Config used

```
techdocs:
  builder: 'local' # Alternatives - 'external'
  generator:
    runIn: local # Alternatives - 'local'
  publisher:
    type: 'awsS3' # Alternatives - 'googleGcs' or 'awsS3'. Read documentation for using alternatives.
    awsS3:
      bucketName: 'techdocs-storage'
      region: mtc
      credentials:
        accessKeyId: xxx
        secretAccessKey: xxx
      endpoint: xxx
      s3ForcePathStyle: true
```

### 🖥️ Your Environment

```
# yarn backstage-cli info
yarn run v1.22.17
$ C:\Work\Experiments\backstage-helm\mrl-backstage\node_modules\.bin\backstage-cli info
OS:   Windows_NT 10.0.19044 - win32/x64
node: v18.12.1
yarn: 1.22.17
cli:  0.22.0 (installed)
backstage:  1.9.1

Dependencies:
  @backstage/app-defaults                          1.0.10
  @backstage/backend-app-api                       0.2.4
  @backstage/backend-common                        0.17.0
  @backstage/backend-plugin-api                    0.2.0
  @backstage/backend-tasks                         0.4.0
  @backstage/backend-test-utils                    0.1.31
  @backstage/catalog-client                        1.2.0
  @backstage/catalog-model                         1.1.4
  @backstage/cli-common                            0.1.11
  @backstage/cli                                   0.22.0
  @backstage/config-loader                         1.1.7
  @backstage/config                                1.0.5
  @backstage/core-app-api                          1.3.0
  @backstage/core-components                       0.12.2
  @backstage/core-plugin-api                       1.2.0
  @backstage/errors                                1.1.4
  @backstage/integration-aws-node                  0.1.0
  @backstage/integration-react                     1.1.8
  @backstage/integration                           1.4.1
  @backstage/plugin-api-docs                       0.8.13
  @backstage/plugin-app-backend                    0.3.39
  @backstage/plugin-auth-backend                   0.17.2
  @backstage/plugin-auth-node                      0.2.8
  @backstage/plugin-azure-devops-backend           0.3.18
  @backstage/plugin-azure-devops-common            0.3.0
  @backstage/plugin-azure-devops                   0.2.4
  @backstage/plugin-catalog-backend-module-azure   0.1.10
  @backstage/plugin-catalog-backend-module-msgraph 0.4.5
  @backstage/plugin-catalog-backend                1.6.0
  @backstage/plugin-catalog-common                 1.0.9
  @backstage/plugin-catalog-graph                  0.2.25
  @backstage/plugin-catalog-import                 0.9.3
  @backstage/plugin-catalog-node                   1.3.0
  @backstage/plugin-catalog-react                  1.2.3
  @backstage/plugin-catalog                        1.7.1
  @backstage/plugin-github-actions                 0.5.13
  @backstage/plugin-kubernetes-backend             0.9.0
  @backstage/plugin-kubernetes-common              0.5.0
  @backstage/plugin-kubernetes                     0.7.6
  @backstage/plugin-org                            0.6.3
  @backstage/plugin-permission-common              0.7.2
  @backstage/plugin-permission-node                0.7.2
  @backstage/plugin-permission-react               0.4.8
  @backstage/plugin-proxy-backend                  0.2.33
  @backstage/plugin-scaffolder-backend             1.9.0
  @backstage/plugin-scaffolder-common              1.2.3
  @backstage/plugin-scaffolder                     1.9.1
  @backstage/plugin-search-backend-module-pg       0.5.0
  @backstage/plugin-search-backend-node            1.1.0
  @backstage/plugin-search-backend                 1.2.0
  @backstage/plugin-search-common                  1.2.0
  @backstage/plugin-search-react                   1.3.1
  @backstage/plugin-search                         1.0.6
  @backstage/plugin-tech-radar                     0.5.20
  @backstage/plugin-techdocs-backend               1.5.0
  @backstage/plugin-techdocs-module-addons-contrib 1.0.8
  @backstage/plugin-techdocs-node                  1.4.3
  @backstage/plugin-techdocs-react                 1.1.1
  @backstage/plugin-techdocs                       1.4.2
  @backstage/plugin-user-settings                  0.6.1
  @backstage/release-manifests                     0.0.8
  @backstage/test-utils                            1.2.3
  @backstage/theme                                 0.2.16
  @backstage/types                                 1.0.2
  @backstage/version-bridge                        1.0.3
```

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by dhhuynh2** [comment]
> ### 👍 Expected behavior
> Should use path style access to S3 if option enabled.
> 
> I believe this is because, in the V3 library the option has changed name from `s3ForcePathStyle` to `forcePathStyle` as documented here https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-s3/interfaces/s3clientconfig.html#forcepathstyle whereas techdocs is using the old property name https://github.com/clareliguori/backstage/blob/cbef804d2375ec9fee75b2cde08397c7bbbcb143/plugins/techdocs-node/src/stages/publish/awsS3.ts#L156

Just wanted to add that I was running into the same issue and tried a patch of `s3ForcePathStyle` to `forcePathStyle` on my instance. Now appears to be working as expected. Can open a PR shortly, if there are no concerns.

**Comment by bralandealmeida** [comment]
> 

Did this patch fix the problem for you? I tried to recreate it but was unsuccessful.

**Comment by bralandealmeida** [comment]
> > ### +1 Expected behavior
> > Should use path style access to S3 if option enabled.
> > I believe this is because, in the V3 library the option has changed name from `s3ForcePathStyle` to `forcePathStyle` as documented here https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-s3/interfaces/s3clientconfig.html#forcepathstyle whereas techdocs is using the old property name https://github.com/clareliguori/backstage/blob/cbef804d2375ec9fee75b2cde08397c7bbbcb143/plugins/techdocs-node/src/stages/publish/awsS3.ts#L156
> 
> Just wanted to add that I was running into the same issue and tried a patch of `s3ForcePathStyle` to `forcePathStyle` on my instance. Now appears to be working as expected. Can open a PR shortly, if there are no concerns.

Did this patch fix the problem for you? I tried to recreate it but was unsuccessful.

**Comment by dhhuynh2** [comment]
> > > ### +1 Expected behavior
> > > Should use path style access to S3 if option enabled.
> > > I believe this is because, in the V3 library the option has changed name from `s3ForcePathStyle` to `forcePathStyle` as documented here https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-s3/interfaces/s3clientconfig.html#forcepathstyle whereas techdocs is using the old property name https://github.com/clareliguori/backstage/blob/cbef804d2375ec9fee75b2cde08397c7bbbcb143/plugins/techdocs-node/src/stages/publish/awsS3.ts#L156
> > 
> > 
> > Just wanted to add that I was running into the same issue and tried a patch of `s3ForcePathStyle` to `forcePathStyle` on my instance. Now appears to be working as expected. Can open a PR shortly, if there are no concerns.
> 
> Did this patch fix the problem for you? I tried to recreate it but was unsuccessful.

Seems to be fixed on our instance of Backstage thus far. I basically applied the changes [in this PR for context](https://github.com/backstage/backstage/pull/15613).

**Comment by bralandealmeida** [comment]
> > > > ### +1 Expected behavior
> > > > Should use path style access to S3 if option enabled.
> > > > I believe this is because, in the V3 library the option has changed name from `s3ForcePathStyle` to `forcePathStyle` as documented here https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-s3/interfaces/s3clientconfig.html#forcepathstyle whereas techdocs is using the old property name https://github.com/clareliguori/backstage/blob/cbef804d2375ec9fee75b2cde08397c7bbbcb143/plugins/techdocs-node/src/stages/publish/awsS3.ts#L156
> > > 
> > > 
> > > Just wanted to add that I was running into the same issue and tried a patch of `s3ForcePathStyle` to `forcePathStyle` on my instance. Now appears to be working as expected. Can open a PR shortly, if there are no concerns.
> > 
> > 
> > Did this patch fix the problem for you? I tried to recreate it but was unsuccessful.
> 
> Seems to be fixed on our instance of Backstage thus far. I basically applied the changes [in this PR for context](https://github.com/backstage/backstage/pull/15613).

I still haven't been successful. I'll wait for the PR merge 🤝

**Comment by dhhuynh2** [comment]
I think this should be fixed in [the latest plugin-techdocs-node, 1.5.0](https://github.com/backstage/backstage/blob/master/plugins/techdocs-node/CHANGELOG.md).

Could you confirm when you get a chance @chris-nolan?
